anitya/lib/backends/crates.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+from anitya.lib.backends import BaseBackend
+from anitya.lib.exceptions import AnityaPluginException
+
+"""
+ © 2017 Igor Gnatenko <ignatenkobrain@fedoraproject.org>
+
+"""
+
+
+class CratesBackend(BaseBackend):
+    ''' The crates class for projects hosted on crates.io. '''
+
+    name = 'crates.io'
+    examples = [
+        'https://crates.io/crates/clap',
+        'https://crates.io/crates/serde',
+    ]
+
+    @classmethod
+    def _get_versions(cls, project):
+        url = 'https://crates.io/api/v1/crates/%s/versions' % project.name
+        try:
+            req = cls.call_url(url)
+        except Exception:  # pragma: no cover
```

Comment:
One day soon it'd probably be good to add all the error handling to the ``call_url`` function so new backends don't have to worry about this, but today is not that day, sadly.

This should catch a less broad exception. Since the URL will never be an FTP URL it'll always go through requests and you can catch an ``requests.RequestException`` for the most general requests error. It would be good to at least include the str() of the exception in the AnityaPluginException message. Something like ``'Could not contact {url}: {reason}'.format(url=url, reason=str(e))``